dist/src/main/java/io/camunda/zeebe/broker/StandaloneBroker.java

```diff
@@ -97,6 +97,7 @@ public void run(final String... args) throws IOException {
   public void onApplicationEvent(final ContextClosedEvent event) {
     try {
       broker.close();
+      brokerClient.close();
```

Comment:
I was also under the impression this was handled by Spring via their bean lifecycle. I'm confused then, as I remember seeing an exception about the actor scheduler being closed twice :thinking: 

Anyway, good catch :+1:

Reply:
:thinking: So this has no impact? https://github.com/camunda/zeebe/blob/8b2238bcbc317907d74ea736b783a361f8b27565/dist/src/main/java/io/camunda/zeebe/broker/BrokerClientConfiguration.java#L33

Reply:
No, this works, I get what the issue :smile: We close the actor scheduler manually, so the broker client cannot be closed, as it hangs (since the actor scheduler is not running). The Spring beans are stopped after the application. If we let Spring manage the actor scheduler as well, I imagine it would figure out the dependency (hopefully? :crossed_fingers:)

Reply:
So we either annotate both `BrokerClientConfiguration` and `ActorSchedulerConfiguration` with the appropriate `Bean(destroyMethod = "close")` and don't close them explicitly or we annotate them both with `Bean(destroyMethod = "")` and close them manually? I guess the first one (letting spring handle it) is somewhat better?

Reply:
I don't know if it's better, because I don't know how well Spring handles inter-bean dependencies (though I think it must do so somewhat well). I'm kind of neutral about it - on one hand Spring is sometimes too much magic for me, on the other, anybody coming in with Spring knowledge would be confused. So...I guess it's a "little" bit better as you said to let Spring handle it, as there's precedent and documentation on how it works, as opposed to us imposing a undocumented convention. But if you both want to do things explicitly I won't oppose :smile:

Reply:
Okay, I've added `destroyMethod` to the atomix cluster, broker client and actor scheduler. This seems to work :shrug:

Reply:
I think it works even if you don't add that, as Spring auto detects your bean is an `AutoCloseable`, which is why I added `destroyMethod = ""` on `AutoCloseable` beans that we were already closing manually. So it's only necessary when it's not a `DisposableBean` or `AutoCloseable`.